### PR TITLE
refactor(cli): update keypair generate command

### PR DESCRIPTION
**Pull request by KaoImin**
## What this PR does / why we need it?

This PR update the axon keypair generator.

### What is the impact of this PR?

No Breaking Change

**PR relation**:
- Ref #

<details><summary>CI Settings</summary><br/>

### **CI Usage**

**Tip**: Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [ ] Web3 Compatible Tests
- [ ] OpenZeppelin tests
- [ ] v3 Core Tests

### **CI Description**

| CI Name                | Description                                                                                             |
| ---------------------- | ------------------------------------------------------------------------------------------------------- |
| *Web3 Compatible Test* | Test the Web3 compatibility of Axon                                                                     |
| *v3 Core Test*         | Run the compatibility tests provided by Uniswap V3                                                      |
| *OpenZeppelin tests*   | Run the compatibility tests provided by OpenZeppelin, including OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19 |
</details>

### CODE REVIEW

**Comment by chaoticlonghair** [comment]
### Questions

Not for your PR, your PR is fine, just ask for future plans.

- Not in CI (or no tests in CI), how to guarantee availability after another PR?
- Is it contained in releases?
  - Why not be a subcommond of Axon? So users don't have to use two binaries?
- Should it contain the peer-id calculation?
  Today, when I generate a new network key, I have to [use CKB to display the peer-id](https://github.com/nervosnetwork/ckb/pull/2114).

**Comment by Flouse** [comment]
> * Should it contain the peer-id calculation?

Do you mean this [peer-id](https://github.com/axonweb3/axon/pull/1621/files#diff-e0d28292a2d348ad9943a03fefd378981094773b3d883f04ccd9e08401588372R56)?

**Comment by chaoticlonghair** [comment]
> > * Should it contain the peer-id calculation?
> 
> Do you mean this [peer-id](https://github.com/axonweb3/axon/pull/1621/files#diff-e0d28292a2d348ad9943a03fefd378981094773b3d883f04ccd9e08401588372R56)?

I reviewed the CLI arguments, if I have a network private key, there is no command to display its info.
Same for BLS key, but network key is more likely to be used, because you have to tell people who want to connect you every time.

If you have more than 1 keys, it's possible you could mix them up.

**Comment by KaoImin** [comment]
> ### Questions
> Not for your PR, your PR is fine, just ask for future plans.
> 
> * Not in CI (or no tests in CI), how to guarantee availability after another PR?
> * Is it contained in releases?
>   
>   * Why not be a subcommond of Axon? So users don't have to use two binaries?
> * Should it contain the peer-id calculation?
>   Today, when I generate a new network key, I have to [use CKB to display the peer-id](https://github.com/nervosnetwork/ckb/pull/2114).

I am integrating the keypair generator into Axon binary.

File: core/cli/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+| name | short | default value |
+|--|--|--|
+| number | n | 4 |
+| private-key-path | p | current_path/free-space/ |
+
+### Update Config and Spec File
+
+Firstly, update the `validator_list` in [chain-spec](../chain/specs/single_node/chain-spec.toml) file. Replace the `bls_pub_key`, `pub_key` and `address` with the value generated by the keypair generator.For example:
+
```
Comment (Flouse) [inline core/cli/README.md]:
```suggestion
First, update the `validator_list` in the [chain-spec](../../devtools/chain/specs/single_node/chain-spec.toml) file. Replace the `bls_pub_key`, `pub_key` and `address` with the value generated by the keypair generator, for example:

```

File: core/cli/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+[[params.verifier_list]]
+bls_pub_key = "0x81ca54f01e3b433c209ae1d165e6362669e13170f89712dbc52c3572901d025ecc206a43c4b25f58388189f74587a7d"
+pub_key = "0x020044def250d1d27c0f7c0eeee6c8b964756d5ad7936fcefaf3f086245ddb9c9c"
+address = "0x83fec4b12b26a70195ccabb67ce0cd692856eaf1"
+propose_weight = 1
+vote_weight = 1
+```
+
+Then, update the `bootstraps` in [config](../chain/config.toml) file. Fill the generated `peer_id` to the `multi_address` For example:
+
```
Comment (Flouse) [inline core/cli/README.md]:
```suggestion
Then, update the `bootstraps` in the [config](../../devtools/chain/config.toml) file. Fill the generated `peer_id` to the `multi_address` For example:

```

File: core/cli/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+```toml
+[[network.bootstraps]]
+multi_address = "/ip4/127.0.0.1/tcp/8001/p2p/QmbZEzvonMiPiioRpYeVWxngjN42FHC3EHXjeo2C7o2NDZ"
+```
+Finally, set the `net_privkey_file` and `bls_privkey_file` with the generated private key file path in [config](../chain/config.toml) file.
+
```
Comment (Flouse) [inline core/cli/README.md]:
```suggestion

Finally, set the `net_privkey_file` and `bls_privkey_file` with the generated private key file path in [config](../../devtools/chain/config.toml) file.

```

File: core/cli/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+# Axon CLI
+
+Axon CLI is a command line tool for Axon framework. It is included in the Axon binary with the command of `cargo build --release`.
+
+## Usage
+
+### Generate Keypair
+
+```bash
+./target/release/axon generate-keypair -n 1
```
Comment (Flouse) [inline core/cli/README.md]:
```suggestion
mkdir -p ./devtools/chain/data
./target/release/axon generate-keypair -n 1 --path ./devtools/chain/data
```

fix `Error writing private key: No such file or directory (os error 2)``

Comment (KaoImin) [reply core/cli/README.md]: The path has the default value `/free-space/` don't need to `mkdir`.

Comment (Flouse) [reply core/cli/README.md]: > fix `Error writing private key: No such file or directory (os error 2)``

I tried the cmd `./target/release/axon generate-keypair -n 1` on a fresh machine, and it threw the above error.

Comment (KaoImin) [reply core/cli/README.md]: Because the folder does not have `free-space` direction.

**Review by Flouse — APPROVED** [review]

**Review by Simon-Tl — APPROVED** [review]

**Pull request merged**